### ISSUE TITLE
R:Create R terminal don't work

### ISSUE THREAD

**Issue by edarague** [Ikuyadeu/vscode-R#11]
When execute the command R:Create R terminal this don't work, the terminal goes blank, is the same for any other command.

My version:

C:\Users\xxxxx>r
R version 3.3.2 (2016-10-31) -- "Sincere Pumpkin Patch"
Copyright (C) 2016 The R Foundation for Statistical Computing
Platform: x86_64-w64-mingw32/x64 (64-bit) ...

**Comment by Ikuyadeu** [comment]
@edarague Thank you for your issue!
Did you set `r.rterm.windows` to your `R.exe` path?
Maybe you should set `C:\\Program Files\\R\\R-3.3.2\\bin\\x64\\R.exe`.

**Comment by edarague** [comment]
Thank you very much, my error was that the version I have installed is different from the one that was in the vs code configurations, so I created a parameter appropriate to my version number in the user settings. In my case v3.3.2

**Comment by skanskan** [comment]
Where do we set r.rterm.windows?
From VS Code?
As a system variable?
or what?

**Comment by Ikuyadeu** [comment]
It's VS Code settings.
1. Please put `F1` key.
2. And type `Preferences: open user settings` or `Ctrl` + `,`
3. You can find `r.rterm.windows`

**Comment by skanskan** [comment]
Thank you.
Finally that method didn't work for me.
Manually editing the file setting.json on the disk.

**Comment by Ikuyadeu** [comment]
I'm not sure, could you restart VS Code?

**Comment by gregkvas** [comment]
@Ikuyadeu thanks for your R Extension for VSCode, eventhough you won't maintain it anymore!

I setted the actual R.exe path for windows via configuring "r.rterm.windows" in the "settings.json" file in the user VSCode settings to {"r.rterm.windows = "C:\\Program Files\\R\\R-3.5.1\\bin\\x64\\R.exe"}, but when trying to run a R code in an opened folder i still get the error message "command 'r.runSelection' not found"!

Does your R-Extension support R up to a specific Version?

Thanks for any help!

**Comment by tmakowski** [comment]
I am having similar issue to @gregkvas. I have built both, VSCode and R, from sources.
My R is located under `/usr/local/src/R/R-3.5.1/bin`. Additionally, I have symlinks to `R` and `Rscript` in `/usr/local/src/R` which is added to `$PATH`. Moreover, I tried solving this as described in #78, however that didn't help.

I have tried manually setting the `Rterm: Linux` and `Rpath: Lsp` to both of the paths I mentioned earlier and that did not help as well. I am still getting the *"command ... not found"*.

**Comment by gregkvas** [comment]
@tmakowski:
1. Did you update VSCode to its newest version?
2. Did you integrate ‘R.exe’ into your Rterm path definition?
3. Did you restart VSCode?

Setting the Rterm-path in VSCode preferences and simply updating VSCode to its newest version (reloading when done) worked for me to solve the issue “command...not found”...

Installing LSP as described in #78 was’nt needed for me to solve the issue, but brought me a great Language Server with ‘hovering-over’ documentation-windows...

**Comment by ohmlovesampere** [comment]
R 3.5.2 and newest VS Code, I still have the problem, and I have set the actual path

**Comment by PalanibharathiM** [comment]
![gitr](https://user-images.githubusercontent.com/41516804/53271279-fc669b80-3713-11e9-8fdb-f0c8333450df.jpg)
@ohmlovesampere check whether the path is correct in settings

**Comment by xiaoqingtan** [comment]
You'll need to open the folder that has R source file.
https://marketplace.visualstudio.com/items?itemName=Ikuyadeu.r

**Comment by vernhow** [comment]
Been awhile but I wanted to provide some help on how I did this, have had error that says "win32 cannot run in windows" and my path was empty or couldn't be found. so here's the steps. 

1. Go to https://cran.r-project.org/bin/windows/base/
2. Select whichever version suits you but for simplicities sake I am on a Windows machine
3. During the runtime installation I unchecked 32-bit and only stayed with the 64-bit as the documentation states there could be a conflict if both are installed at the same time 
4. I had an issue locating my path so I knew my Rx64 GUI 4.0.3 traditional terminal could be invoked 
5. so at the ">" i typed the below
`> R.home()`
you can use this too... 
`> file.path(R.home("bin"), "R")`

**Next Steps:**
1. once you're done locating the path on the typical right clicking on the R.exe icon copy and paste the entire path
2. directly into the VSCode's "R>Rterm: windows" setting field
3. as an example mine was found along with my OneDrive as its a practice I've been doing
4. hope this helps

**Comment by Algosmat** [comment]
That was very helpful

**Comment by iretex** [comment]
> @Ikuyadeu thanks for your R Extension for VSCode, even though you won't maintain it anymore!
> 
> I setted the actual R.exe path for windows via configuring "r.rterm.windows" in the "settings.json" file in the user VSCode settings to {"r.rterm.windows": "C:\Program Files\R\R-3.5.1\bin\x64\R.exe"}, but when trying to run a R code in an opened folder i still get the error message "command 'r.runSelection' not found"!
> 
> Does your R-Extension support R up to a specific Version?
> 
> Thanks for any help!

Thank you so much for this comment, I figured out from my settings.json file that the format that's registered is very bad. All worked fine after fixing up from there.

**Comment by barbosawf** [comment]
> @edarague Thank you for your issue! Did you set `r.rterm.windows` to your `R.exe` path? Maybe you should set `C:\\Program Files\\R\\R-3.3.2\\bin\\x64\\R.exe`.

How can I do that?

**Comment by FaKesk** [comment]
I did everything that you said but it gives me the error "Cannot find R for creating R terminal. Change setting r.rterm.linux to R path.". But I use windows, how can I handle this

**Comment by iretex** [comment]
I used windows as well. Did you edit the path in settings.json file or GUI?

Best regards,
Iretioluwa Olawuyi
+1 437 973 4450
<https://aka.ms/AAb9ysg>
________________________________

**Comment by FaKesk** [comment]
Quoting Iretioluwa Olawuyi ***@***.***>:

> I used windows as well. Did you edit the path in settings.json file or GUI?
>
> Best regards,
> Iretioluwa Olawuyi
> +1 437 973 4450
> <https://aka.ms/AAb9ysg>
> ________________________________
> From: FATMA KESKİN ***@***.***>
> Sent: Thursday, March 30, 2023 2:39:43 PM
> To: REditorSupport/vscode-R ***@***.***>
> Cc: Iretioluwa Olawuyi ***@***.***>; Comment ***@***.***>
> Subject: Re: [REditorSupport/vscode-R] R:Create R terminal don't work (#11)
>
>
> I did everything that you said but it gives me the error "Cannot  
> find R for creating R terminal. Change setting r.rterm.linux to R  
> path.". But I use windows, how can I handle this
>
> —
> Reply to this email directly, view it on  
> GitHub<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FREditorSupport%2Fvscode-R%2Fissues%2F11%23issuecomment-1490754923&data=05%7C01%7C%7Cc59f8c235607446f0a6e08db314e2162%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638157983857568085%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=twk%2B0oHeOJ%2FISYLwPCAoyQY9ADTjbCAfTcuXzQMAeRA%3D&reserved=0>, or  
> unsubscribe<https://emea01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fnotifications%2Funsubscribe-auth%2FAGC2IF2OQ24GEJHVUALBT43W6XHO7ANCNFSM4DPKD3TA&data=05%7C01%7C%7Cc59f8c235607446f0a6e08db314e2162%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C638157983857568085%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=d7Y3cpipJ3Ra2iHsiG2l0AeeUrus6rW3W2oFBIf%2BcGw%3D&reserved=0>.
> You are receiving this because you commented.Message ID: ***@***.***>
>
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/REditorSupport/vscode-R/issues/11#issuecomment-1490916821
> You are receiving this because you commented.
>
> Message ID: ***@***.***>


I changed it in json.settings.

**Comment by kyleGrealis** [comment]
Ubuntu 22.04 user with VSCode 1.84.2 just checking in if anyone has the issue(s) I ran into...
I was following some setup instructions after doing a reinstall on my machine and ran the `R.home()` instructions to get my path. A copy & paste later broke by interactive R terminal, but I hadn't realized it until the next day. It drove me crazy until I reversed what I did. So now my `R > Rterm: Linux` is empty and the world is righteous again.
